src/runtime_src/core/edge/drm/zocl/zocl_dma.h

```diff
@@ -36,7 +59,29 @@ typedef struct zocl_dma_handle {
 	void 			*dma_arg;
 } zocl_dma_handle_t;
 
+/**
+ * zocl_dma_memcpy_pre() - DMA memory copy preparation.
+ *
+ * @dma_handle: zocl dma handler
+ * @dst_paddr: destination dma address
+ * @src_paddr: source dma address
+ * @size: number of bytes to read from source
+ *
+ * Caller get dma_handle from sched_cmd, then get dma_chan from
+ * zocl_dma_channel_instance; caller can optionally set dma_func and dma_arg for
+ * private usage.
```

Comment:
It seems caller does not have option to set dma_func and dma_arg, right? We have to modify the function itself.

Reply:
What about caller can optionally make its own dma_func and dma_arg for private usage. Set implies there is existing dma_func, but actually caller has to make the dma_func and know what is inside the dma_arg.